### DISCUSSION TITLE
We deprecated Java 8 support -- migrate to Java 11 or newer

### DISCUSSION THREAD

**Discussion by eedrummer** [synthetichealth/synthea#1145] [Announcements]
Hello Synthea Community,

The Synthea Team is currently considering revising the minimum version of Java needed to use Synthea. The current minimum version required is Java 8. The proposal is to change the minimum version to Java 11.

## Why make the change?

During a recent Synthea release, the team realized that some of the project dependencies had security vulnerabilities and undertook an effort to update the dependency versions. In this process, the team discovered that some dependencies can't be updated to their latest version due to their minimum Java requirements. Most notable was the [hapi](https://hapifhir.io/hapi-fhir/) library, which is used by Synthea for FHIR export. Staying on Java 8 would prevent us from using hapi to implement FHIR Version R5 export, when that version of the specification passes the ballot at HL7.

## What should you do?

We're posting this announcement because we would like to hear from the community. Particularly if this change would adversely impact you. This could include cases where Synthea is being run in an environment where the version of Java can't be changed. Your feedback can help us decide whether to move forward now or what our maintenance policies should be going forward.

You can also let us know if you would support a move to Java 11 (or even Java 17 if we're getting ambitious).

**Comment by jawalonoski** [comment]
📣 Pull Request #1144 will change the minimum version of Java required to run Synthea to Java 11. ⚠️

We plan to merge this Pull Request within days.

**Comment by jawalonoski** [comment]
Pull Request. #1144 has now been merged, so Synthea now requires a _minimum_ of Java 11.

**Comment by lhs-open** [comment]
Makes sense to move to java 11 or above. Thanks.
